src/core/model_config_utils.cc

```diff
@@ -505,6 +518,7 @@ ValidateModelConfig(
                     std::to_string(gid) + ", valid GPUs are 0 - " +
                     std::to_string(dcnt - 1));
           }
+          RETURN_IF_ERROR(CheckGPUCompatibility(gid));
```

Comment:
Replace dcnt above with GetSupportGPUs and then you can just check gid for membership in the set and if not there print the error that is currently in CheckGPUCompatibility. In general in the normalize and validate routines don't want to replace cudaGetDeviceCount and device_cnt with supported_gpus.